### ISSUE TITLE
Add Weight Decay Schedule

### ISSUE THREAD

**Issue by gkielian** [ReaLLMASIC/nanoGPT#335]
The following paper from NeuriPS 2024 suggests that with a weight decay schedule we can effect ground truth acquisition:

https://arxiv.org/abs/2406.01766

Another paper suggests however that weight decay increase would increase the probability of attention sinks developing:

https://arxiv.org/abs/2410.10781

Would be interesting if better or more semantic position encodings might counter this, or non-normalized softmax alternatives (like the sigmoid or consmax styles), as these should be affected by weight-decay x attentino-sink interaction (which we might want to create a training time metric for...)
